src/backend/drm/compositor/mod.rs

```diff
@@ -1388,6 +1388,11 @@ where
         let current_size = self.output.current_mode().unwrap().size;
         let output_scale = self.output.current_scale().fractional_scale().into();
         let output_transform = self.output.current_transform();
+
+        // Output transform is specified in surface-rotation, so inversion gives us the
+        // render transform for the output itself.
+        let output_transform = output_transform.invert();
```

Comment:
Is this a case of GL's coordinate system being flipped upright and then around the wrong way again? Having to invert the transform seems suspicious.

Reply:
No, as the comment states it's the difference between surface and output rotation.

Reply:
To elaborate: If I turn my phone 90° clockwise, that means the output's rendering needs to be flipped 90° counter-clockwise. However the surface displayed on it get the information that it's flipped clockwise since that's what the protocol specifies.